### PR TITLE
Enable support for Ubuntu 24.04

### PR DESCRIPTION
**Pull request by mc-nv**
This PR enables support for Ubuntu 24.04 by adding a new job to the CI workflow.

### CODE REVIEW

**Review by lkomali — APPROVED** [review]

File: server/install_dependencies_and_build.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ apt update -q=2 \
     && . /etc/os-release \
     && echo "deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ $UBUNTU_CODENAME main" | tee /etc/apt/sources.list.d/kitware.list >/dev/null \
     && apt-get update -q=2 \
-    && apt-get install -y --no-install-recommends cmake=3.27.7* cmake-data=3.27.7* \
+    && apt-get install -y --no-install-recommends cmake=3.28.* cmake-data=3.28.* \
```
Comment (nv-anants) [inline server/install_dependencies_and_build.sh:39]: was this missed on original PR?
we should add patch number here to make it consistent with others

Comment (mc-nv) [reply server/install_dependencies_and_build.sh:39]: Come out with suggestion

Comment (nv-anants) [reply server/install_dependencies_and_build.sh:39]: version should be `cmake=3.28.3*`, similar to other repos - https://github.com/triton-inference-server/server/pull/7758/files
I can push the change, if you prefer that.

Comment (mc-nv) [reply server/install_dependencies_and_build.sh:39]: Let me show you how is suggestion looks: 
```suggestion
    && apt-get install -y --no-install-recommends cmake=3.28.3* cmake-data=3.28.3* \
```

Comment (nv-anants) [reply server/install_dependencies_and_build.sh:39]: looks good, thanks!

**Pull request merged**